src/appkit.cs

```diff
@@ -27594,4 +28119,97 @@ interface NSPreviewRepresentingActivityItem : NSPreviewRepresentableActivityItem
 		NativeHandle Constructor (NSObject item, [NullAllowed] string title, [NullAllowed] NSItemProvider imageProvider, [NullAllowed] NSItemProvider iconProvider);
 	}
 
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSObject))]
+	interface NSNibConnector : NSCoding {
+		[NullAllowed, Export ("source", ArgumentSemantic.Weak)]
+		NSObject Source { get; set; }
+
+		[NullAllowed, Export ("destination", ArgumentSemantic.Weak)]
+		NSObject Destination { get; set; }
+
+		[Export ("label")]
+		string Label { get; set; }
+
+		[Export ("replaceObject:withObject:")]
+		void Replace (NSObject oldObject, NSObject newObject);
+
+		[Export ("establishConnection")]
+		void EstablishConnection ();
+	}
+
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSNibConnector))]
+	interface NSNibControlConnector {
+		[Export ("establishConnection")]
+		void EstablishConnection ();
+	}
+
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSView))]
+	interface NSTextInsertionIndicator {
+		[Export ("displayMode", ArgumentSemantic.Assign)]
+		NSTextInsertionIndicatorDisplayMode DisplayMode { get; set; }
+
+		[NullAllowed, Export ("color", ArgumentSemantic.Copy)]
+		NSColor Color { get; set; }
+
+		[Export ("automaticModeOptions", ArgumentSemantic.Assign)]
+		NSTextInsertionIndicatorAutomaticModeOptions AutomaticModeOptions { get; set; }
+
+		[NullAllowed, Export ("effectsViewInserter", ArgumentSemantic.Copy)]
+		Action<NSView> EffectsViewInserter { get; set; }
+	}
+
+	[NoMacCatalyst, Mac (14, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSMenuItemBadge : NSCopying {
+		[Static]
+		[Export ("updatesWithCount:")]
+		NSMenuItemBadge CreateUpdates (nint itemCount);
+
+		[Static]
+		[Export ("newItemsWithCount:")]
+		NSMenuItemBadge CreateNewItems (nint itemCount);
+
+		[Static]
+		[Export ("alertsWithCount:")]
+		NSMenuItemBadge CreateAlerts (nint itemCount);
+
+		[Export ("initWithCount:type:")]
+		NativeHandle Constructor (nint itemCount, NSMenuItemBadgeType type);
+
+		[Export ("initWithCount:")]
+		NativeHandle Constructor (nint itemCount);
+
+		[Export ("initWithString:")]
+		NativeHandle Constructor (string @string);
+
+		[Export ("itemCount")]
+		nint ItemCount { get; }
+
+		[Export ("type")]
+		NSMenuItemBadgeType Type { get; }
+
+		[NullAllowed, Export ("stringValue")]
+		string StringValue { get; }
+	}
+
+	[Mac (11, 0)]
+	[Protocol]
```

Comment:
base type + model?

Reply:
It does not make sense in this instance if we look at the documentation. Specially when we only have a single property that has to be implemented